### PR TITLE
V3a mdns advertisement

### PR DESCRIPTION
**Pull request by jaredgreene1**
## overview

Adds a new process to the pi's boot script which advertises an http service on port 31591.

This PR includes:
- [X] Chore work

## review requests

Let's try to get this merged asap. With this, we'll be able to push out a v3a image to the pi's and be able to start using the base branch on these machines (which will be awesome!). We have a few moving pieces in the office blocked on this work ATM.

### CODE REVIEW

File: compute/scripts/avahi/server.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import dbus
```
Comment (jaredgreene1) [inline compute/scripts/avahi/server.py]: Add link to resin project

File: compute/scripts/avahi/service.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import dbus
```
Comment (jaredgreene1) [inline compute/scripts/avahi/service.py]: Add link to resin project

File: compute/scripts/dhcp_init.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (jaredgreene1) [inline compute/scripts/dhcp_init.sh]: Remove from this PR

File: compute/scripts/health_server.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import os
```
Comment (jaredgreene1) [inline compute/scripts/health_server.py]: Add TODO for differentiating between wifi and usb (maybe by advertising a slightly different health endpoint on the usb ip vs the wifi ip)

File: compute/scripts/health_server.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            'resin_device_name': os.environ['RESIN_DEVICE_NAME_AT_INIT']
+        }
+
+    return web.json_response(robot_status)
+
+
+
+if __name__ == "__main__":
+    #TODO jmg 11/13/17: advertise an http service
+    local_service = AvahiService("opentrons", "_ssh._tcp", PORT)
```
Comment (jaredgreene1) [inline compute/scripts/health_server.py]: change advertised name to `opentrons-[UUID]` to avoid collisions

File: Dockerfile.template [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
 RUN apt-get update
-RUN apt-get install gcc make git g++
+RUN apt-get install gcc make git g++ vim
+RUN pip install numpy
 
+COPY ./api ./api
+COPY ./compute ./compute
 
-RUN pip install -r ./requirements.txt
+RUN pip install -r ./api/requirements.txt
```
Comment (astaff) [inline Dockerfile.template]: Why not `make -C api install`?

File: compute/scripts/avahi/service.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# https://github.com/resin-io-playground/resin-avahi-dbus/tree/master/avahi
```
Comment (astaff) [inline compute/scripts/avahi/service.py]: What is devision of responsibilities between `server.py` and `service.py`? Can these two be merged together?

File: compute/scripts/announce_mdns.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import os
+
+from avahi.service import AvahiService
+
+
+DEVICE_NAME = os.environ.get('RESIN_DEVICE_NAME_AT_INIT')
+service_name = 'opentrons-{}'.format(DEVICE_NAME)
+AvahiService(service_name, "_http._tcp", port=31951, keep_alive=True)
```
Comment (astaff) [inline compute/scripts/announce_mdns.py]: 31950?

File: compute/scripts/main_start.sh [hunk 9]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket
+
+echo "[BOOT] Starting server"
+. /usr/src/compute/scripts/api_init.sh &
```
Comment (astaff) [inline compute/scripts/main_start.sh]: Call main.py directly?

File: Dockerfile.template (hunk 6 above)
Comment (jaredgreene1) [reply Dockerfile.template]: Note the todo above

File: compute/scripts/announce_mdns.py (hunk 8 above)
Comment (jaredgreene1) [reply compute/scripts/announce_mdns.py]: 👍

**Comment by mcous** [comment]
This might be a good candidate for a manual squash before merge. Thoughts @btmorr and @astaff?

**Pull request merged**